nntrainer/layers/loss/cross_entropy_softmax_loss_layer.cpp

```diff
@@ -85,7 +85,14 @@ void CrossEntropySoftmaxLossLayer::calcDerivative(RunLayerContext &context) {
   /// operation
   // TODO: verify y and ret_derivative must not be same as loss layer is not
   // working in-place
-  ret.subtract(y2, ret_derivative);
+  if (ret.getDataType() != y2.getDataType()) {
+    ret.subtract(y2.clone(ret.getDataType()), ret_derivative);
+  } else {
+    ret.subtract(y2, ret_derivative);
+  }
+
+  applyLossScale(ret_derivative);
```

Comment:
please leave the comment it is necessary to place here to enable mixed precision training.

Reply:
comment is append.